### DISCUSSION TITLE
Help wanted - Intelligent

### DISCUSSION THREAD

**Discussion by BottlecapDave** [BottlecapDave/HomeAssistant-OctopusEnergy#878] [Announcements]
OE have deprecated some of their APIs around intelligent data. I'm wanting to get a head start on it, but as I'm not on intelligent it would be useful to know what the data looks like now.

If you're on intelligent, is there any chance you could run the below graphql queries and provide the results (filling in your account number)?

```
query {
  devices(accountNumber: "A-XXX") {
    id
    deviceType
    provider
    ... on SmartFlexVehicle {
      make
      model
      chargingPreferences {
        weekdayTargetTime
        weekdayTargetSoc
        weekendTargetTime
        weekendTargetSoc
        minimumSoc
        maximumSoc
      }
    }
    ... on SmartFlexChargePoint {
      make
      model
      chargingPreferences {
        weekdayTargetTime
        weekdayTargetSoc
        weekendTargetTime
        weekendTargetSoc
        minimumSoc
        maximumSoc
      }
    }
  }
}
```

Details on running the queries can be found at https://developer.octopus.energy/docs/api/#graphql-api. The query to obtain your auth token is

```
mutation {
	obtainKrakenToken(input: { APIKey: "API_KEY" }) {
		token
	}
}
```

it would be useful to get responses from both people where their charger is in charge and people where their car is in charge.

Thanks!

**Comment by akendall1966** [comment]
Hi Dave,
Saw the request, first time using OE Graphql so hopefully what you're after:

```
{
  "data": {
    "devices": [
      {
        "id": "xxx-xxx-xxx-xxx",
        "deviceType": "ELECTRIC_VEHICLES",
        "provider": "MYENERGI",
        "make": "Myenergi",
        "model": "Zappi smart EV",
        "chargingPreferences": {
          "weekdayTargetTime": "08:00:00",
          "weekdayTargetSoc": 20,
          "weekendTargetTime": "08:00:00",
          "weekendTargetSoc": 20,
          "minimumSoc": null,
          "maximumSoc": null
        }
      }
    ]
  }
}
```

**Reply by BottlecapDave** [reply]
Thank you, that helps :)

**Comment by flip555** [comment]
```
{
  "data": {
    "devices": [
      {
        "id": "xxx-xxx-xxx-xxx",
        "deviceType": "ELECTRIC_VEHICLES",
        "provider": "OHME",
        "make": "Ohme",
        "model": "Ohme home pro charger",
        "chargingPreferences": {
          "weekdayTargetTime": "08:00:00",
          "weekdayTargetSoc": 80,
          "weekendTargetTime": "08:00:00",
          "weekendTargetSoc": 80,
          "minimumSoc": null,
          "maximumSoc": null
        }
      }
    ]
  }
}
```

**Reply by BottlecapDave** [reply]
Thanks for the post

**Comment by ndg63276** [comment]
```
{
  "data": {
    "devices": [
      {
        "id": "xxx-xxx-xxx-xxx-xxx",
        "deviceType": "ELECTRIC_VEHICLES",
        "provider": "OHME",
        "make": "Ohme",
        "model": "Ohme GO Charging Cable Type 2",
        "chargingPreferences": {
          "weekdayTargetTime": "08:00:00",
          "weekdayTargetSoc": 80,
          "weekendTargetTime": "08:00:00",
          "weekendTargetSoc": 80,
          "minimumSoc": null,
          "maximumSoc": null
        }
      }
    ]
  }
}
```
Funnily enough I used to have an Ohme Go Cable, but I switched it to an Ohme ePod recently. I guess that only got updated in the Ohme app, not in Octopus.

**Reply by BottlecapDave** [reply]
Thank you for the post.

**Comment by Cameleon99** [comment]
Formatting all screwy on my phone. 

`{
  "data": {
    "devices": [
      {
        "id": "00000000-0000-0000-0000-00000000A0A0",
        "deviceType": "ELECTRIC_VEHICLES",
        "provider": "MYENERGI",
        "make": "Myenergi",
        "model": "Zappi smart EV",
        "chargingPreferences": {
          "weekdayTargetTime": "07:00:00",
          "weekdayTargetSoc": 15,
          "weekendTargetTime": "07:00:00",
          "weekendTargetSoc": 15,
          "minimumSoc": null,
          "maximumSoc": null
        }
      }
    ]
  }
}`

**Reply by BottlecapDave** [reply]
Thank you for the post.

**Comment by stoogie** [comment]
{
  "data": {
    "devices": [
      {
        "id": "00000000-0001-2000-3456-000000012345",
        "deviceType": "ELECTRIC_VEHICLES",
        "provider": "ENODE",
        "make": "Mini",
        "model": "Electric",
        "chargingPreferences": {
          "weekdayTargetTime": "04:00:00",
          "weekdayTargetSoc": 100,
          "weekendTargetTime": "04:00:00",
          "weekendTargetSoc": 100,
          "minimumSoc": null,
          "maximumSoc": null
        }
      }
    ]
  }
}

**Reply by BottlecapDave** [reply]
Thank you for the post
